diff --git a/location.install b/location.install
index 8f8978d..6f4dc12 100644
--- a/location.install
+++ b/location.install
@@ -962,8 +962,8 @@ function location_update_6303() {
  */
 function location_update_6304() {
   // Skip this update if it was already done on the 5.x side.
-  if (variable_get('location_update_5305_done', FALSE)) {
-    variable_del('location_update_5305_done');
+  if (variable_get('location_5305_done', FALSE)) {
+    variable_del('location_5305_done');
     return array();
   }

@@ -1068,8 +1068,8 @@ function location_update_6305() {
  */
 function location_update_6306() {
   // Skip this update if it was already done on the 5.x side.
-  if (variable_get('location_update_5306_done', FALSE)) {
-    variable_del('location_update_5306_done');
+  if (variable_get('location_5306_done', FALSE)) {
+    variable_del('location_5306_done');
     return array();
   }
   $ret = array();
